bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -85,6 +83,7 @@
         private final EntryLogMetadata entryLogMetadata;
         private final File logFile;
         private long ledgerIdAssigned = UNASSIGNED_LEDGERID;
+        volatile boolean ledgerDirFull = false;
```

Comment:
I am not sure `ledgerDirFull` really belongs to `BufferedLogChannel`. `BufferedLogChannel` is a channel for a log file, which it shouldn't deal with `ledgerDirFull` or not. Can't you put the ledger dir full logic in entry log manager itself?

Reply:
Hey Sijie, I considered about your concern before reintroducing this flag variable here. It would unnecessarily complicate the logic of LedgerDirsListener for maintaining the ledgerdir full status in the entrylogmanager implementation. Thats why I chose to keep it here. 

I don't think it is inappropriate to have it here, since there is final 'logfile' (File) variable for this BufferedLogChannel here, so I would consider 'ledgerDirFull' variable as extension of 'logFile' variable.

Reply:
well if this logic is only to EntryLogPerLedgerManager, it should be maintained in EntryLogPerLedgerManager rather than polluting here. You can wrap BufferedLogChannel with a flag in EntryLogPerLedgerManager if you think you need that flag.

Reply:
Considered that as well, having new class BufferedLogChannelWithDirInfo extending BufferedLogChannelWithDirInfo in EntryLogManagerForEntryLogPerLedger class. But BufferedLogChannel instance is created by EntryLoggerAllocator https://github.com/apache/bookkeeper/blob/master/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLoggerAllocator.java#L162 , so if I create new BufferedLogChannelWithDirInfo class, then here I should have if/else block based on conf value.

is that ok?